### PR TITLE
Add optional selective cache refresh option

### PR DESCRIPTION
**Pull request by sboardwell**
This PR adds the ability to selectively refresh cached elements.

Inspiration taken from https://bluegrid.io/edu/how-to-purge-the-nginx-cache

### CODE REVIEW

File: charts/artifact-caching-proxy/templates/nginx-proxy-configmap.yaml [hunk 1]
```diff
@@ -61,6 +71,10 @@ data:
 
         # Default location for cached requests
         location / {
+{{- if .Values.proxy.proxyBypass.enabled -}}
+            add_header Cache $upstream_cache_status;
```
Comment (sboardwell) [inline charts/artifact-caching-proxy/templates/nginx-proxy-configmap.yaml:75]: The `Cache` header is, of course, optional but I like the transparency it provides.

**Comment by dduportal** [comment]
Thanks @sboardwell ! Are you at ease with helm unitest to add some to cover the code in this change?
Otherwise do you allow me to push a set of unittests in this PR instead?
We can pair if you are interested

**Comment by sboardwell** [comment]
I can add some. Was a little busy today but wanted to get something in for you to look at.

**Comment by dduportal** [comment]
(btw: you have to bump the chart version in order to have the checks validating your PR)

**Comment by sboardwell** [comment]
@dduportal - should be good to go now

File: charts/artifact-caching-proxy/Chart.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ apiVersion: v2
 description: artifact-caching-proxy is a Nginx caching proxy in front of repo.jenkins-ci.org
 name: artifact-caching-proxy
 type: application
-version: 0.12.0
+version: 0.12.1
```
Comment (dduportal) [inline charts/artifact-caching-proxy/Chart.yaml:5]:
```suggestion
version: 0.13.0
```

Comment (dduportal) [reply charts/artifact-caching-proxy/Chart.yaml:5]: Rationale: it's a new feature so minor version bump :)

Comment (sboardwell) [reply charts/artifact-caching-proxy/Chart.yaml]: Yeah, makes sense

**Review by dduportal — APPROVED** [review]
Nice 👍🏼

**Pull request merged**